### DISCUSSION TITLE
column regex split: how to set expansion by default

### DISCUSSION THREAD

**Discussion by aborruso** [saulpw/visidata#2311] [Q&A] [answered]
Hi,
in the previous version of visidata, when I was clicking on `:` to split a column by regex, I had in output the splitted columns.
In this version I must click on `(` to split the output array.

Is there a way to have the automatic split by default? And if so, is there a reason why you would advise against it?

Thank you

**Comment by saulpw** [comment, marked as answer]
You can bind `:` to the `split-col` command, which still exists.  There's no reason I would advise against it; we made the change because it seems more internally consistent.  If it seems like everyone prefers the old way we can certainly change the binding back at any point.  (All scripts will continue to work as they use the longnames.)

**Reply by aborruso** [reply]
> You can bind : to the split-col command

Thank you very much, Now I have to remember how to do it, but don't tell me, I study.

> If it seems like everyone prefers the old way we can certainly change the binding back at any point.

I prefer it, but I'm not everyone 🤣

As usual, thank you very much Saul

**Reply by jonathonbly** [reply]
Thanks so much for mentioning this. 

I was so confused by how column splitting worked. Searching through the docs, I only saw that ':' will split a column on a regex. When I tried that, it looked like all I would get is one new column with semicolons on the regex match. I thought I just did a string substitution. 

When I searched for splitting columns, the results I found didn't mention what the output would be or how to make that output into separate columns. Did I just not spend enough time to understand the docs?

Anyway, thanks again!

**Reply by midichef** [reply]
> > You can bind : to the split-col command
> 
> Thank you very much, Now I have to remember how to do it, but don't tell me, I study.

For future reference, the binding can be changed by adding to `.visidatarc`:
```
Sheet.bindkey(':', 'split-col')
```

> I prefer it, but I'm not everyone 🤣

I also prefer the auto-expand behavior! Long ago I didn't know about `split-col`, so I wrote my own equivalent back then.

**Reply by midichef** [reply]
But `split-col` is deprecated and gives the error `split-col deprecated since v3.0; use a different command`. Is there a way to turn off deprecation in `.visidatarc`?

**Reply by saulpw** [reply]
Hm, I suppose we should undeprecate `split-col`.  It can move out of deprecated.py and remain unbound for the moment.

Maybe `split-col` should be bound to `:` and `addcol-split` bound to `z:`?
